File: Dynamic-Programming/120-M-Triangle/triangle-dp.cpp
class Solution {
public:
    int minimumTotal(vector<vector<int>>& triangle) {
        int sz = triangle.size();
        for (int i=sz-2; i>=0; i--)
        {
            for (int j=0; j<triangle[i].size(); j++)
            {
                triangle[i][j] += std::min(triangle[i+1][j], triangle[i+1][j+1]);
            }
        }
        return triangle[0][0];
    }
};

File: Dynamic-Programming/120-M-Triangle/triangle-dp.java
class Solution {
    public int minimumTotal(List<List<Integer>> triangle) {
        int m = triangle.size();
        for (int i=m-2; i>=0; i--)
        {
            for (int j=0; j<triangle.get(i).size(); j++)
            {
                int minimum = Math.min(triangle.get(i+1).get(j), triangle.get(i+1).get(j+1));
                int val = triangle.get(i).get(j) + minimum;
                triangle.get(i).set(j, val);
            }
        }
        return triangle.get(0).get(0);
    }
}